labgrid/driver/usbstoragedriver.py

```diff
@@ -113,11 +172,26 @@ def write_image(self, filename=None, mode=Mode.DD, partition=None, skip=0, seek=
         )
 
     @Driver.check_active
-    @step(result=True)
-    def get_size(self):
-        args = ["cat", "/sys/class/block/{}/size".format(self.storage.path[5:])]
-        size = subprocess.check_output(self.storage.command_prefix + args)
-        return int(size)*512
+    @step(args=['partition'], result=True)
+    def get_size(self, partition=None):
+        """
+        Get the size of the bound USB storage root device or partition.
+
+        Args:
+            partition (int or None): optional, get size of the specified partition or None for
+                getting the size of the root device (defaults to None)
+
+        Returns:
+            int: size in bytes
+        """
+        partition = "" if partition is None else partition
+        args = ["cat", "/sys/class/block/{}{}/size".format(self.storage.path[5:], partition)]
```

Comment:
Hm, this fails for me right after muxing the SD to the test system:

```
FAILED test_rpi4.py::test_usb_storage_size - subprocess.CalledProcessError: Command '['cat', '/sys/class/block/sdj1/size']' returned non-zero exit status 1.
```
Presumably because the partition table is not read yet?

Reply:
In the current design, `_wait_for_medium` needs to be called first. But since `get_size` is a user callable function, we should add an optional timeout parameter and do the wait inside the function. This will avoid confusion for users calling the function directly, as you did inside the test.

Reply:
We should probably fix this case on the resource level. It does not make much sense to expose a resource as available which is not accessible yet. But this is out of scope for this PR.
Can you provide a small reproducer for this?

Reply:
This is how I can reproduce it, but I guess it depends on the environment (SD muxer..):

```
    d = target.get_driver('USBSDWireDriver')
    d.set_mode('dut')
    time.sleep(1)
    d.set_mode('host')
    storage = target.get_driver('USBStorageDriver')
    # Without one of the two below the error appears.
    #storage._wait_for_medium()
    #time.sleep(2)
    storage.copy_files([ "cmdline2.txt" ])
```

Note that while debugging this I also run into the PTY issue (https://github.com/labgrid-project/labgrid/commit/42f08d5ea32ced0fa8e7993e0fce6f5963a4349e), but with that fixed it is now really just the card reader taking its time until the media is ready.

Reply:
This should be fixed with the latest `USBSDWireDriver`, since it overrides the avail property and only signals availability if both the control and storage device are available. Can you confirm that?

Reply:
I have rebased to the latest master; @agners would you please confirm that it works with the fixed `USBSDWireDriver`?

Reply:
I does not seem like @agners is coming back on this anymore, but I cannot reproduce it either since I do not have a USB-SD-Wire device. However as far as I understand this should have been fixed anyway by now – can we just merge this PR and open an issue if this problem pops up again?